### PR TITLE
Removed camel-connector dependency from component-proxy

### PR DESCRIPTION
**Pull request by valdar**
SwaggerConnectorComponent and OauthRefreshEndpoint do not compile due to remove of camel-connector dependency from component-proxy, they need to be adjusted using component-proxy classes instead.

### CODE REVIEW

File: app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/WithSyndesisHeaderFilterStrategy.java [hunk 1]
```diff
@@ -52,7 +52,7 @@ public boolean applyFilterToCamelHeaders(final String headerName, final Object h
             // false | true | false => global has precedence over rest
             // true | false | true => global has precedence over rest
             // true | true | true => both agree header should be filtered out
-            return globalWouldFilter || globalWouldFilter && !restWouldFilter;
+            return globalWouldFilter || (globalWouldFilter && !restWouldFilter);
```
Comment (valdar) [inline app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/WithSyndesisHeaderFilterStrategy.java]: there is something fishy here, this is equivalent to just return globalWouldFilter;

File: app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/WithSyndesisHeaderFilterStrategy.java [hunk 2]
```diff
@@ -65,7 +65,7 @@ public boolean applyFilterToExternalHeaders(final String headerName, final Objec
...
```
Comment (valdar) [inline app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/WithSyndesisHeaderFilterStrategy.java]: there is something fishy here, this is equivalent to just return globalWouldFilter;

File: app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {
+  "actions": [],
```
Comment (zregvart) [inline app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json]: Generated by `server-api-generator`

File: app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json [hunk 4]
```diff
@@ -23,6 +24,8 @@
       "type": "MAVEN"
     }
   ],
+  "description": "rest-swagger TODO:change this",
```
Comment (zregvart) [inline app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json]: ditto

File: app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json [hunk 5]
```diff
@@ -23,6 +24,8 @@
...
+  "icon": "",
```
Comment (zregvart) [inline app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json]: ditto

File: app/connector/rest-swagger/pom.xml [hunk 6]
```diff
@@ -44,6 +43,16 @@
       <scope>provided</scope>
     </dependency>
 
+    <dependency>
+      <groupId>io.syndesis.common</groupId>
+      <artifactId>common-util</artifactId>
```
Comment (zregvart) [inline app/connector/rest-swagger/pom.xml]: Not too sure about adding this dependency

File: app/connector/rest-swagger/pom.xml [hunk 7]
```diff
@@ -62,6 +71,7 @@
     <dependency>
       <groupId>org.apache.camel</groupId>
       <artifactId>camel-rest-swagger</artifactId>
+      <scope>provided</scope>
```
Comment (zregvart) [inline app/connector/rest-swagger/pom.xml]: Pretty sure that this is a compile time dependency, connector depends on it, it should at least be `runtime`

File: app/connector/support/processor/src/test/java/io/syndesis/connector/support/processor/SyndesisHeaderStrategyTest.java [hunk 8]
```diff
@@ -15,26 +15,31 @@
...
+    Exchange exchange = new DefaultExchange(camelContext);
 
     @Test
     public void shouldFilterOutEverythingButInwardContentTypeHeader() {
         final SyndesisHeaderStrategy headerStrategy = new SyndesisHeaderStrategy();
 
-        assertThat(headerStrategy.applyFilterToCamelHeaders("Content-Type", "", NOT_USED)).isTrue();
-        assertThat(headerStrategy.applyFilterToExternalHeaders("Content-Type", "", NOT_USED)).isFalse();
+        assertThat(headerStrategy.applyFilterToCamelHeaders("Content-Type", "", exchange)).isTrue();
+        assertThat(headerStrategy.applyFilterToExternalHeaders("Content-Type", "", exchange)).isFalse();
```
Comment (zregvart) [inline app/connector/support/processor/src/test/java/io/syndesis/connector/support/processor/SyndesisHeaderStrategyTest.java]: Was this needed?

File: app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json (hunk 3 above)
Comment (valdar) [reply app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json]: it needs to be empty to please the check of connector-support-maven-plugin

File: app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json (hunk 4 above)
Comment (valdar) [reply app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json]: ditto

File: app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json (hunk 5 above)
Comment (valdar) [reply app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json]: ditto

File: app/connector/rest-swagger/pom.xml (hunk 6 above)
Comment (valdar) [reply app/connector/rest-swagger/pom.xml]: PMD was complaining

File: app/connector/rest-swagger/pom.xml (hunk 7 above)
Comment (valdar) [reply app/connector/rest-swagger/pom.xml]: PMD was complaining, but you might be right runtime should suffice

File: app/connector/support/processor/src/test/java/io/syndesis/connector/support/processor/SyndesisHeaderStrategyTest.java (hunk 8 above)
Comment (valdar) [reply app/connector/support/processor/src/test/java/io/syndesis/connector/support/processor/SyndesisHeaderStrategyTest.java]: You wrote the test, I just made it pass :)

**Pull request merged**